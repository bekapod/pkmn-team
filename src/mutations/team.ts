import gql from "graphql-tag";

export const createTeam = gql`
  mutation($name: String!, $pokedexIds: [Int!]!) {
    createTeam(name: $name, pokedexIds: $pokedexIds) {
      id
    }
  }
`;
