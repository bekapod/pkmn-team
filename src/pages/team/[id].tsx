import type {
  GetServerSideProps,
  NextComponentType,
  NextPageContext
} from 'next';
import { useRouter } from 'next/router';
import { initUrqlClient, NextUrqlPageContext, withUrqlClient } from 'next-urql';
import {
  useAllPokemonQuery,
  useTeamByIdQuery,
  useUpdateTeamMutation,
  useDeleteTeamMutation,
  useCreateTeamMembersMutation,
  TeamByIdDocument,
  AllPokemonDocument,
  useDeleteTeamMembersMutation,
  useCreateTeamMemberMovesMutation,
  TeamMemberFragmentFragment,
  useDeleteTeamMemberMovesMutation,
  MoveFragmentFragment
} from '~/generated/graphql';
import { createClient } from '~/lib/client';
import { FullWidthContainer } from '~/components/FullWidthContainer';
import { Page } from '~/components/Page';
import { TeamBuilder } from '~/components/TeamBuilder';
import { useCallback, useMemo, useState } from 'react';
import { ssrExchange } from 'urql';
import isEqual from 'react-fast-compare';

type Props = {
  id?: string;
};

const Team: NextComponentType<
  NextUrqlPageContext & { query: NextPageContext['query'] },
  Props,
  Props
> = ({ id }) => {
  const router = useRouter();
  const teamByIdOptions = useMemo(() => ({ variables: { id }, pause: !id }), [
    id
  ]);
  const [{ data: teamData, fetching: teamFetching }] = useTeamByIdQuery(
    teamByIdOptions
  );

  const [limit] = useState(1000);
  const [offset] = useState(0);
  const allPokemonOptions = useMemo(
    () =>
      ({
        variables: { offset, limit },
        pause: !id
      } as const),
    [id, limit, offset]
  );
  const [{ data: allPokemonData }] = useAllPokemonQuery(allPokemonOptions);

  const [
    { fetching: updateTeamFetching },
    updateTeam
  ] = useUpdateTeamMutation();
  const [
    { fetching: deleteTeamFetching },
    deleteTeam
  ] = useDeleteTeamMutation();
  const [
    { fetching: createTeamMembersFetching },
    createTeamMembers
  ] = useCreateTeamMembersMutation();
  const [
    { fetching: deleteTeamMembersFetching },
    deleteTeamMembers
  ] = useDeleteTeamMembersMutation();
  const [
    { fetching: createTeamMemberMovesFetching },
    createTeamMemberMoves
  ] = useCreateTeamMemberMovesMutation();
  const [
    { fetching: deleteTeamMemberMovesFetching },
    deleteTeamMemberMoves
  ] = useDeleteTeamMemberMovesMutation();

  const team = teamData?.teamById ?? undefined;
  const pokemon = allPokemonData?.pokemon;

  const updateTeamHandler = useCallback(
    (name: string) => {
      if (team?.id && team?.name !== name) {
        updateTeam({ id: team?.id, name });
      }
    },
    [team?.id, team?.name, updateTeam]
  );

  const deleteTeamHandler = useCallback(async () => {
    if (team?.id) {
      const { data } = await deleteTeam({ id: team?.id });
      if (data?.deleteTeam?.id) router.push('/');
    }
  }, [team?.id, router, deleteTeam]);

  const updateTeamMembersHandler = useCallback(
    (members: TeamMemberFragmentFragment[]) => {
      const membersToUpdate = members.map(({ id, pokemon, order }) => ({
        id,
        pokemon_id: pokemon.id,
        order,
        team_id: team?.id
      }));

      const membersToDelete = team?.team_members
        .filter(({ id }) => !members.find(newMember => id === newMember.id))
        .map(({ id }) => id);

      if (membersToUpdate.length && !isEqual(members, team?.team_members)) {
        createTeamMembers({
          members: membersToUpdate
        });
      }

      if (membersToDelete && membersToDelete.length) {
        deleteTeamMembers({
          members: membersToDelete
        });
      }
    },
    // eslint-disable-next-line react-hooks/exhaustive-deps
    [JSON.stringify(team), createTeamMembers]
  );

  const updateTeamMemberMovesHandler = useCallback(
    (member: TeamMemberFragmentFragment, moves: MoveFragmentFragment[]) => {
      const movesToDelete = member?.learned_moves
        ?.filter(({ move }) => !moves.find(newMove => move.id === newMove.id))
        .map(({ move }) => move.id);
      const remainingLearnedMoves = member?.learned_moves
        ?.filter(({ move }) => moves.find(newMove => move.id === newMove.id))
        .map(({ move }) => move);

      console.log({
        moves,
        learnedMoves: remainingLearnedMoves,
        isEqual: isEqual(moves, remainingLearnedMoves),
        movesToDelete
      });

      if (moves.length && !isEqual(moves, remainingLearnedMoves)) {
        console.log('CREATING');
        createTeamMemberMoves({
          moves: moves.map((move, idx) => ({
            move_id: move.id,
            order: idx,
            team_member_id: member.id
          }))
        });
      }

      if (movesToDelete && movesToDelete.length) {
        console.log('DELETING');
        deleteTeamMemberMoves({
          moveIds: movesToDelete,
          memberId: member.id
        });
      }
    },
    [createTeamMemberMoves, deleteTeamMemberMoves]
  );

  return (
    <Page
      title={`Team: ${team?.name ?? '…'}`}
      metaTitle={`${team?.name ?? '…'} | Team | Pkmn Team`}
    >
      <FullWidthContainer>
        <TeamBuilder
          allPokemon={pokemon}
          team={team}
          isSkeleton={teamFetching}
          isLoading={
            updateTeamFetching ||
            deleteTeamFetching ||
            createTeamMembersFetching ||
            deleteTeamMembersFetching ||
            createTeamMemberMovesFetching ||
            deleteTeamMemberMovesFetching
          }
          updateTeam={updateTeamHandler}
          deleteTeam={deleteTeamHandler}
          updateTeamMembers={updateTeamMembersHandler}
          updateTeamMemberMoves={updateTeamMemberMovesHandler}
        />
      </FullWidthContainer>
    </Page>
  );
};

export const getServerSideProps: GetServerSideProps = async context => {
  const ssrCache = ssrExchange({ isClient: false });
  const client = initUrqlClient(
    createClient(`${process.env.INTERNAL_GRAPHQL_ENDPOINT}/graphql`)(ssrCache),
    false
  );
  const id = context.params?.id?.toString();

  if (client && id) {
    await Promise.all([client.query(TeamByIdDocument, { id }).toPromise()]);
  }

  return {
    props: {
      urqlState: ssrCache.extractData(),
      id
    }
  };
};

export default withUrqlClient(createClient(), { ssr: false })(Team);
