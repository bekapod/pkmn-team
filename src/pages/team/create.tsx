import { useRouter } from 'next/router';
import { useEffect } from 'react';
import type { NextComponentType } from 'next';
import { NextUrqlPageContext, withUrqlClient } from 'next-urql';
import { createClient } from '~/lib/client';
import { FullWidthContainer } from '~/components/FullWidthContainer';
import { Page } from '~/components/Page';
import { TeamCreator, TeamCreatorFormValues } from '~/components/TeamCreator';
import { useCreateTeamMutation } from '~/generated/graphql';

const TeamCreate: NextComponentType<NextUrqlPageContext> = () => {
  const router = useRouter();
  const [{ data, fetching }, createTeam] = useCreateTeamMutation();

  useEffect(() => {
    const id = data?.createTeam?.id;
    if (id) {
      router.push(`/team/${id}`);
    }
  }, [data?.createTeam?.id, router]);

  const createTeamHandler = (values: TeamCreatorFormValues) => {
    createTeam({ name: values['team-name'] });
  };

  return (
    <Page title="Create a team" metaTitle="Create a team | Team | Pkmn Team">
      <FullWidthContainer>
        <TeamCreator createTeam={createTeamHandler} isLoading={fetching} />
      </FullWidthContainer>
    </Page>
  );
};

export default withUrqlClient(createClient(), { ssr: false })(TeamCreate);
