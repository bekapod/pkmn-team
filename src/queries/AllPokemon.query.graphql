query AllPokemon {
  allPokemonOrderedByPokedexId {
    before
    after
    data {
      name
      slug
      pokedexId
      sprite
      types {
        name
        slug
        pokemon {
          name
        }
      }
      learnableMoves {
        version
        learnMethod
        levelLearnedAt
        move {
          name
          slug
          type {
            name
            slug
          }
        }
      }
    }
  }
}
