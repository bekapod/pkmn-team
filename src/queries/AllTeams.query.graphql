query AllTeams {
  teams(order_by: { created_at: desc_nulls_last }) {
    id
    name
    created_at
    team_members {
      id
      order
      pokemon {
        id
        pokedex_id
        name
        slug
        sprite
        types {
          type_id
          type {
            id
            name
            slug
          }
        }
      }
    }
  }
}
