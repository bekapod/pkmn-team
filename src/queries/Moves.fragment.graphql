fragment MoveFragment on moves {
  id
  slug
  name
  accuracy
  power
  pp
  damage_class {
    value
  }
  type {
    ...TypeFragment
  }
}
