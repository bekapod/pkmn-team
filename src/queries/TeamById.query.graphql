query TeamById($id: uuid!) {
  teamById(id: $id) {
    id
    name
    created_at
    team_members {
      id
      order
      pokemon {
        id
        pokedex_id
        name
        slug
        sprite
        types {
          type_id
          type {
            id
            name
            slug
          }
        }
      }
    }
  }
}
