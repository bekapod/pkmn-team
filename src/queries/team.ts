import gql from "graphql-tag";

export const getAllTeams = gql`
  query {
    allTeams {
      id
      name
      createdAt
      members {
        id
        pokemon {
          pokedexId
          name
          types
          sprite
        }
      }
    }
  }
`;
