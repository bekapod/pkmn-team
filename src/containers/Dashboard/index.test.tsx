import React from "react";
import { MockedProvider, MockedResponse } from "react-apollo/test-utils";
import { MemoryRouter } from "react-router-dom";
// tslint:disable-next-line:no-implicit-dependencies
import renderer from "react-test-renderer";
// tslint:disable-next-line:no-implicit-dependencies
import wait from "waait";
import DashboardContainer from ".";
import { getAllTeams } from "../../queries/team";

const mocks: ReadonlyArray<MockedResponse> = [
  {
    request: {
      query: getAllTeams
    },
    result: {
      data: {
        allTeams: [
          {
            createdAt: "2018-06-08T21:15:14.723Z",
            id: "cji6gz8gwhblk0a9639btq2hd",
            members: [
              {
                id: "cji6gz8gwhbll0a96aahx3ivv",
                pokemon: {
                  name: "bulbasaur",
                  pokedexId: 1,
                  sprite: "1.png",
                  types: ["POISON", "GRASS"]
                }
              },
              {
                id: "cji6gz8gwhblm0a96eja18t10",
                pokemon: {
                  name: "charmander",
                  pokedexId: 4,
                  sprite: "4.png",
                  types: ["FIRE"]
                }
              },
              {
                id: "cji6gz8gwhbln0a96q7wmx9zj",
                pokemon: {
                  name: "squirtle",
                  pokedexId: 7,
                  sprite: "7.png",
                  types: ["WATER"]
                }
              },
              {
                id: "cji6gz8gwhblo0a96wgoki379",
                pokemon: {
                  name: "pikachu",
                  pokedexId: 25,
                  sprite: "25.png",
                  types: ["ELECTRIC"]
                }
              }
            ],
            name: "Starters Team"
          },
          {
            createdAt: "2018-06-08T23:06:11.936Z",
            id: "cji6kxx7gp5n80a96mgw7gc45",
            loading: false,
            members: [
              {
                id: "cji6kxx7hp5n90a96r5rpx50u",
                pokemon: {
                  name: "pikachu",
                  pokedexId: 25,
                  sprite: "25.png",
                  types: ["ELECTRIC"]
                }
              },
              {
                id: "cji6kxx7hp5na0a96hb6e0643",
                pokemon: {
                  name: "pikachu",
                  pokedexId: 25,
                  sprite: "25.png",
                  types: ["ELECTRIC"]
                }
              },
              {
                id: "cji6kxx7hp5nb0a9629wdve5s",
                pokemon: {
                  name: "pikachu",
                  pokedexId: 25,
                  sprite: "25.png",
                  types: ["ELECTRIC"]
                }
              },
              {
                id: "cji6kxx7hp5nc0a969th9ll1f",
                pokemon: {
                  name: "pikachu",
                  pokedexId: 25,
                  sprite: "25.png",
                  types: ["ELECTRIC"]
                }
              },
              {
                id: "cji6kxx7hp5nd0a968aj6f6qj",
                pokemon: {
                  name: "pikachu",
                  pokedexId: 25,
                  sprite: "25.png",
                  types: ["ELECTRIC"]
                }
              },
              {
                id: "cji6kxx7hp5ne0a964tf72wol",
                pokemon: {
                  name: "pikachu",
                  pokedexId: 25,
                  sprite: "25.png",
                  types: ["ELECTRIC"]
                }
              }
            ],
            name: "Pikachu Team"
          },
          {
            createdAt: "2018-06-08T23:07:46.587Z",
            id: "cji6kzy8pprbq0a964rc3ikk3",
            members: [
              {
                id: "cji6kzy8pprbr0a967jhcik0n",
                pokemon: {
                  name: "pikachu",
                  pokedexId: 25,
                  sprite: "25.png",
                  types: ["ELECTRIC"]
                }
              },
              {
                id: "cji6kzy8pprbs0a96349zeq46",
                pokemon: {
                  name: "pikachu",
                  pokedexId: 25,
                  sprite: "25.png",
                  types: ["ELECTRIC"]
                }
              },
              {
                id: "cji6kzy8pprbt0a96v5isjwvt",
                pokemon: {
                  name: "pikachu",
                  pokedexId: 25,
                  sprite: "25.png",
                  types: ["ELECTRIC"]
                }
              }
            ],
            name: "Pikachu Team"
          }
        ]
      }
    }
  }
];

describe("<DashboardContainer />", () => {
  it("renders without crashing", async () => {
    const tree = renderer.create(
      <MockedProvider mocks={mocks} addTypename={false}>
        <MemoryRouter initialEntries={["/"]}>
          <DashboardContainer />
        </MemoryRouter>
      </MockedProvider>
    );

    await wait(0);

    expect(tree.toJSON()).toMatchSnapshot();
  });
});
