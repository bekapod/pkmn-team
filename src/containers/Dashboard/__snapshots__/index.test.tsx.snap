// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DashboardContainer /> renders without crashing 1`] = `
Array [
  <h1
    className="Heading-b3adig-0 jwQDLL"
  >
    My Teams
  </h1>,
  <div
    className="SectionContainer-xjlkvu-0 jZbaLj"
  >
    <div
      className="CenteredRow-f3rx2f-0 hNpdbY"
    >
      <a
        className="Cta__CtaInternalLink-h8nk1r-0 jNQDYG"
        href="/team/create/"
        onClick={[Function]}
      >
        Create a team
      </a>
    </div>
    <div
      className="TeamGrid__CardGrid-sc-12cfyys-0 dpqImA"
    >
      <article
        className="Card__CardWrapper-sc-3o7tcb-0 bTjMad"
      >
        <header
          className="Card__CardHeader-sc-3o7tcb-1 dzDCfA"
        >
          <h2
            className="CardHeading-mmeo34-0 ebVIiR"
          >
            Starters Team
          </h2>
        </header>
        <div
          className="Card__CardContent-sc-3o7tcb-2 fDtTnd"
        >
          <div
            className="CardMeta__CardMetaWrapper-yqfxft-0 kTYlws"
          >
            <div
              className="CardMeta__CardMetaItem-yqfxft-1 hPoDTe"
            >
              <span
                className="CardMeta__CardMetaLabel-yqfxft-2 hiNCFT"
              >
                Pkmn
              </span>
              <span
                className="CardMeta__CardMetaValue-yqfxft-3 jqYdyo"
              >
                4
              </span>
            </div>
            <div
              className="CardMeta__CardMetaItem-yqfxft-1 hPoDTe"
            >
              <span
                className="CardMeta__CardMetaLabel-yqfxft-2 hiNCFT"
              >
                Created
              </span>
              <span
                className="CardMeta__CardMetaValue-yqfxft-3 jqYdyo"
              >
                8/6/18
              </span>
            </div>
          </div>
          <div
            className="PokemonLine__Row-sc-1qiwepa-0 kdjbDA"
          >
            <img
              alt="bulbasaur sprite"
              className="PokemonLine__RowImage-sc-1qiwepa-1 hUcFnr"
              src="/sprites/1.png"
            />
            <div>
              <div
                className="PokemonLine__RowTitle-sc-1qiwepa-2 edSfRl"
              >
                #1 Bulbasaur
              </div>
              <ul
                className="InlineList-sc-1dn69qz-0 YzWxJ"
              >
                <li>
                  <span
                    className="TypeTag-sc-1dttu8y-0 jHkiSl"
                    type="GRASS"
                  >
                    GRASS
                  </span>
                </li>
                <li>
                  <span
                    className="TypeTag-sc-1dttu8y-0 jjNEEF"
                    type="POISON"
                  >
                    POISON
                  </span>
                </li>
              </ul>
            </div>
          </div>
          <div
            className="PokemonLine__Row-sc-1qiwepa-0 pbtXk"
          >
            <img
              alt="charmander sprite"
              className="PokemonLine__RowImage-sc-1qiwepa-1 hUcFnr"
              src="/sprites/4.png"
            />
            <div>
              <div
                className="PokemonLine__RowTitle-sc-1qiwepa-2 edSfRl"
              >
                #4 Charmander
              </div>
              <ul
                className="InlineList-sc-1dn69qz-0 YzWxJ"
              >
                <li>
                  <span
                    className="TypeTag-sc-1dttu8y-0 QKQcU"
                    type="FIRE"
                  >
                    FIRE
                  </span>
                </li>
              </ul>
            </div>
          </div>
          <div
            className="PokemonLine__Row-sc-1qiwepa-0 fvpxMP"
          >
            <img
              alt="squirtle sprite"
              className="PokemonLine__RowImage-sc-1qiwepa-1 hUcFnr"
              src="/sprites/7.png"
            />
            <div>
              <div
                className="PokemonLine__RowTitle-sc-1qiwepa-2 edSfRl"
              >
                #7 Squirtle
              </div>
              <ul
                className="InlineList-sc-1dn69qz-0 YzWxJ"
              >
                <li>
                  <span
                    className="TypeTag-sc-1dttu8y-0 hRaMMw"
                    type="WATER"
                  >
                    WATER
                  </span>
                </li>
              </ul>
            </div>
          </div>
          <div
            className="PokemonLine__Row-sc-1qiwepa-0 dAGLvI"
          >
            <img
              alt="pikachu sprite"
              className="PokemonLine__RowImage-sc-1qiwepa-1 hUcFnr"
              src="/sprites/25.png"
            />
            <div>
              <div
                className="PokemonLine__RowTitle-sc-1qiwepa-2 edSfRl"
              >
                #25 Pikachu
              </div>
              <ul
                className="InlineList-sc-1dn69qz-0 YzWxJ"
              >
                <li>
                  <span
                    className="TypeTag-sc-1dttu8y-0 byzsso"
                    type="ELECTRIC"
                  >
                    ELECTRIC
                  </span>
                </li>
              </ul>
            </div>
          </div>
        </div>
      </article>
      <article
        className="Card__CardWrapper-sc-3o7tcb-0 bTjMad"
      >
        <header
          className="Card__CardHeader-sc-3o7tcb-1 eyrCfh"
        >
          <h2
            className="CardHeading-mmeo34-0 ebVIiR"
          >
            Pikachu Team
          </h2>
        </header>
        <div
          className="Card__CardContent-sc-3o7tcb-2 fDtTnd"
        >
          <div
            className="CardMeta__CardMetaWrapper-yqfxft-0 kTYlws"
          >
            <div
              className="CardMeta__CardMetaItem-yqfxft-1 hPoDTe"
            >
              <span
                className="CardMeta__CardMetaLabel-yqfxft-2 hiNCFT"
              >
                Pkmn
              </span>
              <span
                className="CardMeta__CardMetaValue-yqfxft-3 jqYdyo"
              >
                6
              </span>
            </div>
            <div
              className="CardMeta__CardMetaItem-yqfxft-1 hPoDTe"
            >
              <span
                className="CardMeta__CardMetaLabel-yqfxft-2 hiNCFT"
              >
                Created
              </span>
              <span
                className="CardMeta__CardMetaValue-yqfxft-3 jqYdyo"
              >
                8/6/18
              </span>
            </div>
          </div>
          <div
            className="PokemonLine__Row-sc-1qiwepa-0 dAGLvI"
          >
            <img
              alt="pikachu sprite"
              className="PokemonLine__RowImage-sc-1qiwepa-1 hUcFnr"
              src="/sprites/25.png"
            />
            <div>
              <div
                className="PokemonLine__RowTitle-sc-1qiwepa-2 edSfRl"
              >
                #25 Pikachu
              </div>
              <ul
                className="InlineList-sc-1dn69qz-0 YzWxJ"
              >
                <li>
                  <span
                    className="TypeTag-sc-1dttu8y-0 byzsso"
                    type="ELECTRIC"
                  >
                    ELECTRIC
                  </span>
                </li>
              </ul>
            </div>
          </div>
          <div
            className="PokemonLine__Row-sc-1qiwepa-0 dAGLvI"
          >
            <img
              alt="pikachu sprite"
              className="PokemonLine__RowImage-sc-1qiwepa-1 hUcFnr"
              src="/sprites/25.png"
            />
            <div>
              <div
                className="PokemonLine__RowTitle-sc-1qiwepa-2 edSfRl"
              >
                #25 Pikachu
              </div>
              <ul
                className="InlineList-sc-1dn69qz-0 YzWxJ"
              >
                <li>
                  <span
                    className="TypeTag-sc-1dttu8y-0 byzsso"
                    type="ELECTRIC"
                  >
                    ELECTRIC
                  </span>
                </li>
              </ul>
            </div>
          </div>
          <div
            className="PokemonLine__Row-sc-1qiwepa-0 dAGLvI"
          >
            <img
              alt="pikachu sprite"
              className="PokemonLine__RowImage-sc-1qiwepa-1 hUcFnr"
              src="/sprites/25.png"
            />
            <div>
              <div
                className="PokemonLine__RowTitle-sc-1qiwepa-2 edSfRl"
              >
                #25 Pikachu
              </div>
              <ul
                className="InlineList-sc-1dn69qz-0 YzWxJ"
              >
                <li>
                  <span
                    className="TypeTag-sc-1dttu8y-0 byzsso"
                    type="ELECTRIC"
                  >
                    ELECTRIC
                  </span>
                </li>
              </ul>
            </div>
          </div>
          <div
            className="PokemonLine__Row-sc-1qiwepa-0 dAGLvI"
          >
            <img
              alt="pikachu sprite"
              className="PokemonLine__RowImage-sc-1qiwepa-1 hUcFnr"
              src="/sprites/25.png"
            />
            <div>
              <div
                className="PokemonLine__RowTitle-sc-1qiwepa-2 edSfRl"
              >
                #25 Pikachu
              </div>
              <ul
                className="InlineList-sc-1dn69qz-0 YzWxJ"
              >
                <li>
                  <span
                    className="TypeTag-sc-1dttu8y-0 byzsso"
                    type="ELECTRIC"
                  >
                    ELECTRIC
                  </span>
                </li>
              </ul>
            </div>
          </div>
          <div
            className="PokemonLine__Row-sc-1qiwepa-0 dAGLvI"
          >
            <img
              alt="pikachu sprite"
              className="PokemonLine__RowImage-sc-1qiwepa-1 hUcFnr"
              src="/sprites/25.png"
            />
            <div>
              <div
                className="PokemonLine__RowTitle-sc-1qiwepa-2 edSfRl"
              >
                #25 Pikachu
              </div>
              <ul
                className="InlineList-sc-1dn69qz-0 YzWxJ"
              >
                <li>
                  <span
                    className="TypeTag-sc-1dttu8y-0 byzsso"
                    type="ELECTRIC"
                  >
                    ELECTRIC
                  </span>
                </li>
              </ul>
            </div>
          </div>
          <div
            className="PokemonLine__Row-sc-1qiwepa-0 dAGLvI"
          >
            <img
              alt="pikachu sprite"
              className="PokemonLine__RowImage-sc-1qiwepa-1 hUcFnr"
              src="/sprites/25.png"
            />
            <div>
              <div
                className="PokemonLine__RowTitle-sc-1qiwepa-2 edSfRl"
              >
                #25 Pikachu
              </div>
              <ul
                className="InlineList-sc-1dn69qz-0 YzWxJ"
              >
                <li>
                  <span
                    className="TypeTag-sc-1dttu8y-0 byzsso"
                    type="ELECTRIC"
                  >
                    ELECTRIC
                  </span>
                </li>
              </ul>
            </div>
          </div>
        </div>
      </article>
      <article
        className="Card__CardWrapper-sc-3o7tcb-0 bTjMad"
      >
        <header
          className="Card__CardHeader-sc-3o7tcb-1 kYOeEy"
        >
          <h2
            className="CardHeading-mmeo34-0 ebVIiR"
          >
            Pikachu Team
          </h2>
        </header>
        <div
          className="Card__CardContent-sc-3o7tcb-2 fDtTnd"
        >
          <div
            className="CardMeta__CardMetaWrapper-yqfxft-0 kTYlws"
          >
            <div
              className="CardMeta__CardMetaItem-yqfxft-1 hPoDTe"
            >
              <span
                className="CardMeta__CardMetaLabel-yqfxft-2 hiNCFT"
              >
                Pkmn
              </span>
              <span
                className="CardMeta__CardMetaValue-yqfxft-3 jqYdyo"
              >
                3
              </span>
            </div>
            <div
              className="CardMeta__CardMetaItem-yqfxft-1 hPoDTe"
            >
              <span
                className="CardMeta__CardMetaLabel-yqfxft-2 hiNCFT"
              >
                Created
              </span>
              <span
                className="CardMeta__CardMetaValue-yqfxft-3 jqYdyo"
              >
                8/6/18
              </span>
            </div>
          </div>
          <div
            className="PokemonLine__Row-sc-1qiwepa-0 dAGLvI"
          >
            <img
              alt="pikachu sprite"
              className="PokemonLine__RowImage-sc-1qiwepa-1 hUcFnr"
              src="/sprites/25.png"
            />
            <div>
              <div
                className="PokemonLine__RowTitle-sc-1qiwepa-2 edSfRl"
              >
                #25 Pikachu
              </div>
              <ul
                className="InlineList-sc-1dn69qz-0 YzWxJ"
              >
                <li>
                  <span
                    className="TypeTag-sc-1dttu8y-0 byzsso"
                    type="ELECTRIC"
                  >
                    ELECTRIC
                  </span>
                </li>
              </ul>
            </div>
          </div>
          <div
            className="PokemonLine__Row-sc-1qiwepa-0 dAGLvI"
          >
            <img
              alt="pikachu sprite"
              className="PokemonLine__RowImage-sc-1qiwepa-1 hUcFnr"
              src="/sprites/25.png"
            />
            <div>
              <div
                className="PokemonLine__RowTitle-sc-1qiwepa-2 edSfRl"
              >
                #25 Pikachu
              </div>
              <ul
                className="InlineList-sc-1dn69qz-0 YzWxJ"
              >
                <li>
                  <span
                    className="TypeTag-sc-1dttu8y-0 byzsso"
                    type="ELECTRIC"
                  >
                    ELECTRIC
                  </span>
                </li>
              </ul>
            </div>
          </div>
          <div
            className="PokemonLine__Row-sc-1qiwepa-0 dAGLvI"
          >
            <img
              alt="pikachu sprite"
              className="PokemonLine__RowImage-sc-1qiwepa-1 hUcFnr"
              src="/sprites/25.png"
            />
            <div>
              <div
                className="PokemonLine__RowTitle-sc-1qiwepa-2 edSfRl"
              >
                #25 Pikachu
              </div>
              <ul
                className="InlineList-sc-1dn69qz-0 YzWxJ"
              >
                <li>
                  <span
                    className="TypeTag-sc-1dttu8y-0 byzsso"
                    type="ELECTRIC"
                  >
                    ELECTRIC
                  </span>
                </li>
              </ul>
            </div>
          </div>
        </div>
      </article>
    </div>
  </div>,
]
`;
